ksp_plugin/interface_collision.cpp

```diff
@@ -0,0 +1,162 @@
+#include "ksp_plugin/interface.hpp"
+
+#include <memory>
+#include <utility>
+
+#include "base/push_pull_callback.hpp"
+#include "journal/method.hpp"
+#include "journal/profiles.hpp"  // 🧙 For generated profiles.
+#include "ksp_plugin/frames.hpp"
+#include "physics/degrees_of_freedom.hpp"
+#include "quantities/quantities.hpp"
+#include "quantities/si.hpp"
+
+namespace principia {
+namespace interface {
+
+using namespace principia::base::_push_pull_callback;
+using namespace principia::journal::_method;
+using namespace principia::ksp_plugin::_frames;
+using namespace principia::physics::_degrees_of_freedom;
+using namespace principia::quantities::_quantities;
+using namespace principia::quantities::_si;
+
+namespace {
+
+template<typename TrajectoryLike>
+not_null<std::unique_ptr<
+    PushPullExecutor<DegreesOfFreedom<World>, Length, Angle, Angle>>>
+NewExecutor(Plugin const* const plugin,
+            int const celestial_index,
+            XYZ const sun_world_position,
+            TrajectoryLike const& vessel_trajectory,
+            double const apoapside_time,
+            double const periapside_time) {
+  CHECK_NOTNULL(plugin);
+  auto const begin =
+      vessel_trajectory.lower_bound(FromGameTime(*plugin, apoapside_time));
+  auto const end =
+      vessel_trajectory.lower_bound(FromGameTime(*plugin, periapside_time));
```

Comment:
This makes it impossible to satisfy the precondition. Make ComputeAndRenderCollision take times.